jetty-server/src/main/java/org/eclipse/jetty/server/handler/ContextHandlerCollection.java

```diff
@@ -58,6 +58,7 @@
 
     @Deprecated
     private Class<? extends ContextHandler> _contextClass = ContextHandler.class;
+    private volatile Handler _singleContext;
```

Comment:
Instead of maintaining an extra variable, you could instead test on `mapping.getHandlers().length == 1` and shortcut to `mapping.getHandlers()[0]`.

Reply:
That's what I did first up, then thought this was simpler.... but your way is probably less cost for those that do have multiple contexts, so I will revert

Reply:
@lorban done

Reply:
For the record I suggested getting rid of the extra field not for efficiency, but for simplicity. :-)